dataproxy/service.go

```diff
@@ -87,6 +91,54 @@ func (s Service) CreateUploadLocation(ctx context.Context, req *service.CreateUp
 	}, nil
 }
 
+// CreateDownloadLink retrieves the requested artifact type for a given execution (wf, node, task) as a signed url(s).
+func (s Service) CreateDownloadLink(ctx context.Context, req *service.CreateDownloadLinkRequest) (
+	resp *service.CreateDownloadLinkResponse, err error) {
+	if req, err = s.validateCreateDownloadLinkRequest(req); err != nil {
+		return nil, err
+	}
+
+	// Lookup task, node, workflow execution
+	var nativeURL string
+	switch o := req.GetSource().(type) {
+	case *service.CreateDownloadLinkRequest_TaskId:
+		return nil, fmt.Errorf("not implemented for task")
+	case *service.CreateDownloadLinkRequest_NodeId:
+		node, err := s.nodeExecutionManager.GetNodeExecution(ctx, admin.NodeExecutionGetRequest{
+			Id: o.NodeId,
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to find node execution [%v]. Error: %w", o.NodeId, err)
```

Comment:
(Not directly related to your change since this seems to be a common pattern in dataproxy service) Any reason not to use grpc status errors and specialized codes here?